### ISSUE TITLE
JSON exception when deserializing Stream Analytics messages in Azure Service Bus

### ISSUE THREAD

**Issue by ahamidou** [Azure/azure-stream-analytics#51]
Stream Analytics isn't properly serializing it's json messages when the output is Azure Service Bus.
The json string always starts with :
<pre>
@\u0006string\b3http://schemas.microsoft.com/2003/10/Serialization/�h
</pre>

UTF8 message sample body preview:
![image](https://user-images.githubusercontent.com/6707600/45324993-b333de00-b51d-11e8-866e-80a069a8d319.png)

I tried to find a solution for this issue but I wasn't, the only thing I found was an issue posted in 2016 with no fix so far: [Service Bus Topics Connector JSON message issue](https://social.msdn.microsoft.com/Forums/azure/en-US/16eb3183-3d16-4fcb-8657-11a50aec1657/service-bus-topics-connector-json-message-issue?forum=azurelogicapps)

**Is there a work around, am I missing something here, or is it simply an existing issue?**

For the moment, I have implemented this work around:
``` csharp
private static T Parse<T>(Message message)
{
    var body = Encoding.UTF8.GetString(message.Body);
    try
    {
        return JsonConvert.DeserializeObject<T>(body);                
    }
    catch (JsonReaderException re)
    {
        if (!re.Message.Contains("Unexpected character encountered while parsing value")) throw;
        var firstArrayIndex = body.IndexOf("[", StringComparison.Ordinal);
        var jsonString = body.Substring(firstArrayIndex < 0
            ? body.IndexOf("{", StringComparison.Ordinal)
            : firstArrayIndex);
        return JsonConvert.DeserializeObject<T>(jsonString);
    }
}

```

**Comment by sidramadoss** [comment]
@ahamidou - this has been addressed in compat level 1.1. On portal, go to 'Compatibility level' under Configure and select 1.1. Let us know if you still face this issue after making this change.

**Comment by ahamidou** [comment]
Lifesaver... Thank you @sidramadoss !
